base/error.jl

```diff
@@ -85,8 +85,21 @@ function _reformat_bt(bt::Array{Ptr{Cvoid},1}, bt2::Array{Any,1})
         tag       = (entry_metadata >> 6) & 0xf
         header    =  entry_metadata >> 10
         if tag == 1 # JL_BT_INTERP_FRAME_TAG
-            code = bt2[j]::Union{CodeInfo,Core.MethodInstance,Nothing}
-            mod = njlvalues == 2 ? bt2[j+1]::Union{Module,Nothing} : nothing
+            code = bt2[j]::Union{CodeInfo,Method,Core.MethodInstance,Nothing}
+            mod = nothing
+            if isa(code, Method)
+                # This is an opaque closure that ran in the interpreter, so
+                # never had a MethodInstance synthesized for it. Do that now,
+                # so we can show something sensible in backtraces
+                mi = ccall(:jl_new_method_instance_uninit,
+                    Ref{Core.MethodInstance}, ())
```

Comment:
I think generally we assume MethodInstance objects are type-uniqued (to make them convenient as IdDict keys), which this internal constructor method bypasses and therefore violates in assumption (particularly since people often may like to attempt to serialize backtraces–which is also possibly risky for that and other related reasons)

Reply:
Ok, this was just for convenience, we can create some other object.